x-pack/plugins/discover_log_explorer/public/state_machines/integrations/src/state_machine.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { assign, createMachine } from 'xstate';
+import { isEmpty, omitBy } from 'lodash';
+import { createComparatorByField } from '../../../utils/comparator_by_field';
+import { Dataset, Integration } from '../../../../common/datasets';
+import { FindIntegrationsResponse } from '../../../../common/latest';
+import { IDatasetsClient } from '../../../services/datasets';
+import { DEFAULT_CONTEXT } from './defaults';
+import {
+  DefaultIntegrationsContext,
+  IntegrationsContext,
+  IntegrationsEvent,
+  IntegrationsSearchParams,
+  IntegrationTypestate,
+} from './types';
+
+export const createPureIntegrationsStateMachine = (
+  initialContext: DefaultIntegrationsContext = DEFAULT_CONTEXT
+) =>
+  createMachine<IntegrationsContext, IntegrationsEvent, IntegrationTypestate>(
+    {
+      /** @xstate-layout N4IgpgJg5mDOIC5QEkB2AXMUBOBDdAlgPaqwB0ArqgdYbgDYEBekAxANoAMAuoqAA5FYBQiT4gAHogCMAJmkB2Mp1kKAzGukBWDQE5ZWgDQgAnol0AWC2QAc0-Vtm7tANheOAvh+NpMOfMSkZPREuBA0UKwQJGBkNABuRADWsb5YeKJBIWERCAlEAMYBJFzcpeKCwpniUggAtHJkljpyRqbmCjZk9hb2LtIWLjZqshZePhjpxVmh4aiRYNjYRNhk-PT4AGYrALZkaf6Z5NlzUHmoiUWZpeVIIJUigTUy0tKcZGpaipwKWgrNFjUxjMCBcFi0ZEBAwsqhsqk4al04xABwygWOs0gcQg9DArAAMgB5ACCABEAPoAWUJACUAKLk5AAOQAKnSAOI04ks5CEpkAZVuAiEjzEd1qiNkZE6XxcPxcul0nRcwMQkqaFhsnE4Ni+StkLmRqOmGLCWIIOLx-LpxJpAGEABKM1kcrk8vmCngVEXVcWIA1S2R2EZqTpvA0KVUIOG6aWfYYWfQKBT2I2TQ7o4KYiDY3Gsfm0lnOtmc7m8gVC+4+p5+hAG94uX42cGyTiK9S6KNglzdThQ8HDA1gtN+NEkU0Qc2W-M2+1O5klt3l-nk-ks+nEyme3h3B6+0C1OoGNSQmxwhSNtTa+SAqPyKUDThDTRuaRa2SyEdTI5Zs05i15gWNJFgurplh6q7rjaW6VnuNYHog-y2Fouihm0IKjNIZAuBomoaAiF5PmM3goumY4zH+v6nJSKx4tEqCxPkKT7GRJpUViJwRDR2BgOclzTDcXq7tWYoIQgiJdBYF7uMqzYGjYKrtAg4JKJwridAoUlfGoX4ZuO7E5px8zcXiizLKs6xbLsLGjmxJwcbMXG0XxhQCTwsEiagzzRroEIxlowzaFoXxRloT4aloUK-G+gzERMtk-vZOaTgARkQVAFGA-JgLg2AFAAFhExpHDOtqOsWYHuhWQnClU8GSIgfwQm+vwBfKWpqVGmldI2PwqP0mnyAounkROWKpelqCZdluUFUVrElRIsDoPgsS4JsmDYAAFFenAAJSsMVmZJWQE0ZVlOV5YV8xHeOHl1aJDUIKhWHBR+bhaEMrzuFGMbdAoPzJp9-wKnFpEJcd2anWAaXnTNV3zRD478ugPG4DssClXOFWllVK5rhuME1VWD1ebWdQKpCuhyh+iY2CmF5RoitiNq2YbNs2Ulg7dFGTslMOTdNl1zTdC3oijaMY6wS0rZgZDrZtO3agdPNjfzsNTRds3XVAqsSzlGP3aKZNiXUIyxkqbYtW8-z0xYTNqU0YL6KzPyONII12Y58wAGK4AQuIQKw9JEmSONLh6Rv7k9rTWO4LgGihuoKtIXWathAMAwaigwooXgkagRCTvAdyq96pPeXUMIQpbzj0zbuh21GDSfN0Iz-PIdhfMFOkkarlDULQBAMMwkDl8b3mvM43Rgk+BhRuGWHfW2OpyMGHt92L+lGVA4-R7UryKGQwVSa22ruJqXWKmQH5XsGnB-BJnuJdme-1YeoZKLX1s-I3mnN1JbC7d9Bvm0EFA0z9IaUQAmAN+j0D4KRvoMHU1NfI2EVGeJmF4PiKh+OgkYOEGaQO3lDHeJk4EmyemCE8uoe6oUGLoJ8nYlJSSwgpVsYVqEIgGMQ3m40BZw2FjrMuwkK61mCu8TS-R0FaV6AqX6fZ-o-DPmeGEqheFq2hhrIW2tEbfnFqjA2JdaoT3Jm8HsBhkyJlBpYemikQRvGsLIK8nCdC2M+p+TeSNeYRD9gHMeojTFiUPjQ6Q7g4RqVznIKM4JYxfCsBYFQdM+z5w8EAA */
```

Comment:
The layout of the machine was a bit unreadable for me, so I tried to improve it.

```suggestion
      /** @xstate-layout N4IgpgJg5mDOIC5QEkB2AXMUBOBDdAlgPaqwB0ArqgdYbgDYEBekAxANoAMAuoqAA5FYBQiT4gAHogC0ATgBs8sgEYAHAHYALMs7rV85ctkAmADQgAnonWcysgKwHZmgMwPlN1aoC+382kwcfGJSMnoiXAgaKFYIEjAyGgA3IgBrBICsPFFQ8MjohGSiAGNgki5uCvFBYRzxKQRpdQ8yY3tlRTVOF05OTXMrBGUXJXtZVU1J1UMXTWNffwwsstyIqNQYsGxsImwyfnp8ADNdgFsyTKCc8jz1qELUFNKciqqkEBqREPqZHWNjMj2GzGTiqWQuYyuYyqAaIaFKVScZzydT2EY6FwuBYgS7ZEI3NaQRIQehgVgAGQA8gBBAAiAH0ALKUgBKAFF6cgAHIAFTZAHEWdSechKVyAMpvARCL5id4NaTGHqApyyDzKYxaDywhA9VRkMGycGcZSuPSg7G4lYEyJEggksnitnUlkAYQAEpzeQKhSKxZKeNUZXV5TIXOpZGRNA5-prUUigTq2uoyLNOGjFJoI7N5n4cUsrviwoSIMTSaxxayeV6+YLhaKJVKPsHvqHGs1bBrFJj7JN7I4zJZEJolOpjApZoZ5FD9JaC3iSDaIHaHRXnW7Pdza76G+L6eKeezqYyA7x3p8Q6AFSCEUZe2jVD1ZEidUZNGQI2iO7IbA5ZHPAgXVZbVLe1y0rFlqy3H1639fdD2dE8mwvVsrzDMYozRVx7FUTUxiRdQdWkQwAUMbQRkhZ8XCBTQAOWa5ixAxi7kZXYyTiVAEiKdILnna1mKJW5olY7AwAeJ4VleQNzxbOU0IQLN3wotxphsFFDCI2YDXsOM0Q1I1NVovMrQY25BLWYS2NYLYdj2A5jjOXjAP4szSyEjYRLEopnhCKSz2lWpUMkX4TXfdN-lw0E+kUQihwQHD9W6XRsOmEdXDowtFwE0tlwAIyIKhijAcUwFwbBigAC2iEz8TXF0PRrGC-UbaSAtlVAfkaexbAhdRUQmU1+yBLQdTmWx0xwmxPFmeRc0WZzTJLMg8oK1AipKsrKuqvjrlYCRYHQfAElwI5MGwAAKHpOAASlYGqstc5awHywritK8qqo2e7SGQ2SOrbaRpxTVR2jGXDUXBPRRuMZRAX7EH1HkBR1BcU0jPm+ii0elbXo2j7toW-FxXQUTcFOWA6o3Rq62avcDyPJDWubQK5OC+Leg-VHDA8MdXDUeQdQhSNUY7cdmkxTg5vzQmHqWnG1rezbPqgb7YGJ0nyb2g6jrIE6zsu3pbtV7KnpehW8a2r6dqJknSvJ36Wf++Sug-BR2mMeRerBadRv1f4FDVT3wV69Hpcx2X8g2AAxXACFJCBWHZKk6Wpnd-Qd9rOuIzVYemadOBREHoxRpMQdTJEzRBTUwXUXw81QIhl3gd5VaDR2s+GIwDQ6TQC70XsfxcIjoRTMiOg1XCDAjUPjaoGgvgYZhIDbzOAYMsh0xHboNF7Nwh7i4j7FTMYPcfftZhRtEMqApdohXy82eI7pR2jExEb1D3YsGDpbD6xFnH7DoUE4Zr4uRLPfIKCo5gpmjDpf4fVuoOC-r8I0ZADDaFmvoHQyh2hYmMtbCOy5QIOggaza8XhATTFwtMGMIM2g6iRphWayU+4I3sKAxaTF3JQE8qQp2bMXYozcECHsU5lClyPm4I0mh8LdC0FLY22NnqrXWu9S2KsCGkD4VncMsNwxImfkiDQfVBYKDhglac05wQKA4VjOWyjcZqOVqrdWdtm5tQfg0DUAI5jB17moOY44BZxXVK0Bw1EcH6D5v+fBMtgJ3BjnHZeMl24A2wZGHQ4If5omrsg3UiNWh9RjPpaM4467eCAA */
```